### PR TITLE
Add count down argument to OSS::Button component

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?

Inspired of destructive button component, a new argument is added to `OSS::Button` named `countDown`. It takes two key:
- callback : function to execute at the end for the count
- time: the time to wait before execution callback

Related to: Cooldown candidate list

### What are the observable changes?
Peek 2022-03-23 17-36
Screenshot from 2022-03-23 17-33-26

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/button.ts [hunk 1]
```diff
@@ -125,8 +140,43 @@ export default class OSSButton extends Component<ButtonArgs> {
...
+      this.intervalID = setInterval(this.updateCounter, DEFAULT_STEP_COUNTER_TIME, this);
+    }
+  }
+
+  updateCounter(ctx: any): void {
+    if (ctx.counterTimeLeft > DEFAULT_STEP_COUNTER_TIME) {
+      ctx.counterTimeLeft -= DEFAULT_STEP_COUNTER_TIME;
+    } else {
+      ctx.clearInterval(ctx.intervalID);
+      ctx.args.countDown?.callback();
```
[the remark this answers is not in the document]
Comment (JulienVannier66) [reply addon/components/o-s-s/button.ts:173]: You right good catch

File: addon/components/o-s-s/button.hbs [hunk 2]
```diff
@@ -1,5 +1,7 @@
-<button type="button" class={{this.computedClass}} {{did-insert this.didInsert}} ...attributes>
-  {{#if this.loadingState}}
+<button type="button" class={{this.computedClass}} {{did-insert this.didInsert}} {{on "click" this.onclick}} ...attributes>
```
[the remark this answers is not in the document]
Comment (JulienVannier66) [reply addon/components/o-s-s/button.hbs:1]: The goal is to generalize the `destructive-button` with all skin and use one component

File: addon/components/o-s-s/button.ts [hunk 3]
```diff
@@ -125,8 +140,43 @@ export default class OSSButton extends Component<ButtonArgs> {
...
+
+    if (this.intervalID) {
+      this.clearInterval(this.intervalID);
+    } else {
+      this.counterTimeLeft = this.args.countDown.time || DEFAULT_COUNTER_TIME;
+      this.intervalState = true;
+
+      this.intervalID = setInterval(this.updateCounter, DEFAULT_STEP_COUNTER_TIME, this);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (JulienVannier66) [reply addon/components/o-s-s/button.ts:166]: No I don't think so, do you have an example?

File: translations/en-us.yaml [hunk 4]
```diff
@@ -9,3 +9,5 @@ oss-components:
     empty_img_alt: Empty content
   password-input:
     regex_error: 'Your password should have at least 8 characters with at least one lower-case character, one upper-case character and one digit.'
+  button:
+    cancel_message: Click to Cancel {time}s...
```
Comment (phndiaye) [inline translations/en-us.yaml:13]: "Click to cancel"

File: addon/components/o-s-s/button.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tgallice) [reply addon/components/o-s-s/button.ts:166]: Nop I just not sure on how current button are used ^^

**Review by tgallice — APPROVED** [review]

**Pull request merged**